src/devices/cpu/arcompact/arcompact.h

```diff
@@ -105,795 +40,751 @@ class arcompact_device : public cpu_device
+	}
+
+	uint8_t common32_get_p(uint32_t op)
+	{
+		return (op & 0x00c00000) >> 22;
+	}
+
+	uint8_t common32_get_areg_reserved(uint32_t op)
+	{
+		return (op & 0x0000003f) >> 0;
+	}
+
+	uint32_t common32_get_s12(uint32_t op)
+	{
+		int S_temp = (op & 0x0000003f) >> 0;
+		int s_temp = (op & 0x00000fc0) >> 6;
+		uint32_t S = s_temp | (S_temp<<6);
+
+		if (S & 0x800) // sign extend
+			S |= 0xfffff000;
+
+		return S;
+	}
+
+	uint32_t common32_get_u6(uint32_t op)
+	{
+		return (op & 0x00000fc0) >> 6;
+	}
+
+	int check_limm16(uint8_t hreg)
+	{
+		if (hreg == REG_LIMM)
+		{
+			get_limm_16bit_opcode();
+			return 6;
+		}
+		return 2;
+	}
+
+	int check_limm(uint8_t reg)
+	{
+		if (reg == REG_LIMM)
+		{
+			get_limm_32bit_opcode();
+			return 8;
+		}
+		return 4;
+	}
+
+	int check_limm(uint8_t breg, uint8_t creg)
+	{
+		if ((breg == REG_LIMM) || (creg == REG_LIMM))
+		{
+			get_limm_32bit_opcode();
+			return 8;
+		}
+		return 4;
+	}
+
+	uint8_t group_0e_get_h(uint16_t op)
+	{
+		uint8_t h = ((op & 0x0007) << 3);
+		h |= ((op & 0x00e0) >> 5);
+		return h;
+	}
+
+	uint8_t common32_get_condition(uint32_t op)
+	{
+		return op & 0x0000001f;
+	}
+
+	void status32_set_e1(void) { m_status32 |= E1_FLAG; }
+	void status32_clear_e1(void) { m_status32 &= ~E1_FLAG; }
+	bool status32_check_e1(void) { return (m_status32 & E1_FLAG ? true : false); }
+
+	void status32_set_e2(void) { m_status32 |= E2_FLAG; }
+	void status32_clear_e2(void) { m_status32 &= ~E2_FLAG; }
+	bool status32_check_e2(void) { return (m_status32 & E2_FLAG ? true : false); }
+
+	// V = overflow (set if signed operation would overflow)
+	void status32_set_v(void) { m_status32 |= V_OVERFLOW_FLAG; }
+	void status32_clear_v(void) { m_status32 &= ~V_OVERFLOW_FLAG; }
+	bool status32_check_v(void) { return (m_status32 & V_OVERFLOW_FLAG ? true : false); }
+
+	// C = carry (unsigned op, carry set is same condition as LO Lower Than, carry clear is same condition as HS Higher Same)
+	void status32_set_c(void) { m_status32 |=  C_CARRY_FLAG; }
+	void status32_clear_c(void) { m_status32 &= ~C_CARRY_FLAG; }
+	bool status32_check_c(void) { return (m_status32 &   C_CARRY_FLAG ? true : false); }
+
+	// N = negative (set if most significant bit of result is set)
+	void status32_set_n(void)  { m_status32 |=  N_NEGATIVE_FLAG; }
+	void status32_clear_n(void) { m_status32 &= ~N_NEGATIVE_FLAG; }
+	bool status32_check_n(void) { return (m_status32 &   N_NEGATIVE_FLAG ? true : false); }
+
+	// Z = zero (set if result is zero, ie both values the same for CMP)
+	void status32_set_z(void) { m_status32 |=  Z_ZERO_FLAG; }
+	void status32_clear_z(void) { m_status32 &= ~Z_ZERO_FLAG; }
+	bool status32_check_z(void) { return (m_status32 &   Z_ZERO_FLAG ? true : false); }
+
+	// debug is the name of the register, this is not a debug function
+	void debugreg_set_ZZ(void) { m_debug |= (1<<23); }
+	void debugreg_clear_ZZ(void) { m_debug &= ~(1<<23); }
+	bool debugreg_check_ZZ(void) { return (m_debug & (1<<23)) ? true : false; }
```

Comment:
Please use empty parentheses consistently for functions with no arguments.

Reply:
done